asdfghjkl/operations/linear.py

```diff
@@ -44,6 +44,14 @@ def cov_kron_A(module, in_data):
     def cov_kron_B(module, out_grads):
         return torch.matmul(out_grads.T, out_grads)  # f_out x f_out
 
+    @staticmethod
+    def cov_unit_wise(module, in_data, out_grads):
+        if module.bias is not None:
+            in_data = torch.cat([in_data, torch.ones(in_data.size(0), 1)], 1)
+        in_in = torch.matmul(in_data.T, in_data)  # f_in x f_in
+        grad_grad = out_grads.mul(out_grads).sum(dim=0)  # f_out
+        return torch.einsum('i, jk -> ijk', grad_grad, in_in).to(in_data.device)
+
```

Comment:
This logic seems not correct.

This is the unit-wise block-diagonal Fisher for the l-th layer

![image](https://user-images.githubusercontent.com/7961228/137265189-2b888729-4b51-42bf-87eb-4de0cf651288.png)

And each diagonal block is defined like this:

![image](https://user-images.githubusercontent.com/7961228/137265319-82fd0257-fff8-49d1-ac64-d190ea0d2893.png)

For now, just ignore `\mathbb{E}`. 
`M_{l-1}` is the input dimension of the l-th layer.
`M_l` is the output dimension of the l-th layer.
`<>` represents a data average.
`h` is the layer input (i.e., `in_data`)
`\nabla_u \ell` is the gradient w.r.t. layer output (i.e., `out_grads`) 

as you can see, `in_data` and `out_grads` "for each example" need to be multiplied by each other before taking an data average.